src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -151,49 +151,50 @@ private void AddSource(string fileName, string source, bool isRootContextDef = f
                 string? @namespace = _currentContext.ContextType.Namespace;
                 bool isInGlobalNamespace = @namespace == JsonConstants.GlobalNamespaceValue;
 
-                StringBuilder sb = new(@"// <auto-generated/>
+                StringBuilder sb = new("""
+// <auto-generated/>
 
 #nullable enable annotations
 #nullable disable warnings
 
 // Suppress warnings about [Obsolete] member usage in generated code.
-#pragma warning disable CS0618");
+#pragma warning disable CS0618
+
+
+""");
+                int indentation = 0;
```

Comment:
FWIW, we use IndentedTextWriter for this purpose in the regex generator.

Reply:
Nice. I think we definitely need to overhaul the formatting tools we're using here.